backend/frontend/tests/test_page_users_and_teams_nav.py

```diff
@@ -121,3 +122,112 @@ def test_user_sees_teams_that_match_the_search_term(self):
         self.page.wait_for_load_state("networkidle")
 
         expect(found_teams).to_contain_text("No teams found")
+
+
+class TestUsers(FrontendTestMixin):
+    def _setup_user_with_github(self):
+        self.user_with_github = UserFactory(
+            first_name="Sam",
+            last_name="Boha",
+            email="sam@rpg.boom"
+        )
+        SocialProfileFactory(user=self.user_with_github, github_name="samboha_codes")
+
+    def _setup_user_without_github(self):
+        self.user_without_github = UserFactory(
+            first_name="John",
+            last_name="Doe",
+            email="john@doe.email"
+        )
+
+    def setUp(self):
+        super().setUp()
+        self.user = UserFactory()
+        self.user.set_password(self.user.email)
+        self.user.save()
+        
+        self.superuser = UserFactory(is_superuser=True)
+        self._setup_user_with_github()
+        self._setup_user_without_github()
+        
+    def _goto_users_page(self):
+        url = self.reverse_url("users_and_teams_nav")
+        self.page.goto(url)
+        self.page.wait_for_load_state()
+    
+
+    def test_all_users_show_up_onload(self):
+        self.do_login(self.user)
+        self._goto_users_page()
+
+        self.page.wait_for_load_state("networkidle")
+
+        users_div = self.page.locator("#users")
+        for user in [self.user, self.user_without_github, self.user_with_github]:
+            expect(users_div).to_contain_text(user.get_full_name())
```

Comment:
Do we need to spin up a whole browser for all of these tests? 

And what happens if we have 1000 users? We shouldn't expect them all to be displayed.